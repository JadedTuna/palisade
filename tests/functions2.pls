in {
    high secret: bool;
}
out {
    low exfil: bool;
}

fn foo() int {
    throw;
    return 42;
}

exfil = false;
try {
    if (secret) {
        z := foo();
    }
} catch {
    exfil = true;
}
