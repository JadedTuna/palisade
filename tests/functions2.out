[1;35msecurity error: [0mthrow in high context is not allowed
   7 | 
   8 | fn foo() int {
   9 |     [1;35mthrow[0m;
       [1;35m~~~~[0m[1;35m^^^^^[0m
